api/v1/helpers/verbs/doPatch.js

```diff
@@ -31,30 +33,38 @@ const httpStatus = require('../../constants').httpStatus;
 function doPatch(req, res, next, props) {
   const resultObj = { reqStartTime: new Date() };
   const requestBody = req.swagger.params.queryBody.value;
-  u.findByKey(props, req.swagger.params)
-  .then((o) => u.isWritable(req, o,
-    featureToggles.isFeatureEnabled('enforceWritePermission')))
-  .then((o) => {
-
-    // To avoid timeouts when patching samples; force the update, even if
-    // the value has not changed. Adding this to the "before update hook" does
-    // give the needed effect; so adding it here!!!.
-    if (props.modelName === 'Sample') {
-      o.changed('value', true);
+  let patchPromise;
+  if (featureToggles.isFeatureEnabled(constants.featureName) &&
+   props.modelName === 'Sample') {
+    const rLinks = requestBody.relatedLinks;
+    if (rLinks) {
+      u.checkDuplicateRLinks(rLinks);
```

Comment:
Oh sorry, I missed this, Thanks for pointing out. I have this test failing if I remove this:
```
  1) api: redisStore: PATCH /v1/samples Patch Related Links  with duplicate name:
     Uncaught AssertionError: expected { Object (status, statusChangedAt, ...) } to have a property 'errors'
      at Test.api.patch.set.send.end (tests/cache/models/samples/patch.js:196:34)
      at Test.assert (node_modules/supertest/lib/test.js:156:6)
      at Server.assert (node_modules/supertest/lib/test.js:127:12)
      at emitCloseNT (net.js:1548:8)
      at _combinedTickCallback (internal/process/next_tick.js:71:11)
      at process._tickCallback (internal/process/next_tick.js:98:9)
```
This check is done in sequelize models using validateJsonSchema function, so it is sequelize error when redisStore is not enabled.